src/Symfony/Component/Security/Core/Authentication/Token/Storage/UsageTrackingTokenStorage.php

```diff
@@ -72,7 +73,28 @@ public function disableUsageTracking(): void
     public static function getSubscribedServices(): array
     {
         return [
-            'session' => SessionInterface::class,
+            'request_stack' => RequestStack::class,
         ];
     }
+
+    private function getSession(): SessionInterface
+    {
+        // BC for symfony/security-bundle < 5.3
+        if ($this->container->has('session')) {
+            trigger_deprecation('symfony/security-core', '5.3', 'Injecting the `session` in "%s" is deprecated, inject the "request_stack" instead.', __CLASS__);
+
+            return $this->container->get('session');
+        }
+
+        if (method_exists($requestStack = $this->container->get('request_stack'), 'getSession')) {
```

Comment:
this should not be needed: the request_stack service should be passed only when it's compatible

Reply:
This breaks compatibility with http-foundation < 5.3.

Is it OK to add a conflict rule in composer.json?

Reply:
fixed